conda/cli/conda_argparse.py

```diff
@@ -376,7 +377,7 @@ def configure_parser_config(sub_parsers):
         help="""Write to the given file. Otherwise writes to the user config file ({user})
 or the file path given by the 'CONDARC' environment variable, if it is set
 (default: %(default)s).""".format(user=escaped_user_rc_path),
-        default=os.environ.get('CONDARC', escaped_user_rc_path)
+        default=os.environ.get('CONDARC', user_rc_path).replace("%", "%%")
```

Comment:
Just `default=os.environ.get('CONDARC', user_rc_path)`. (Any included `%`s won't be replaced as that replacement is already done for `%(default)s`. Also we want `default` to have the actual value.)

Reply:
I think we're seeing the error differently.  The error I'm trying to avoid is

```
  File "E:\Miniconda3\lib\argparse.py", line 606, in _expand_help
    return self._get_help_string(action) % params
```

If the contents of `CONDARC` are something like `c:\Users\dwerp\%USERPROFILE%`, then I think we still have problems without the escaping.

Reply:
```
Python 3.5.2 (default, Sep 15 2016, 07:38:42)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> "%USERPROFILE% %s" % 'hey'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: unsupported format character 'U' (0x55) at index 1
>>> "%%USERPROFILE%% %s" % 'hey'
'%USERPROFILE% hey'
>>>
```

Reply:
because https://github.com/conda/conda/issues/6645#issuecomment-360146809 ¯\\\_(ツ)\_/¯

Reply:
In `argparse` the help string formatting (with the default formatter class) is done only once, hence for the snippet you pointed at we get something equivalent to
```python
x = self._get_help_string(action)
x == r"""Write to the given file. Otherwise writes to the user config file (c:\Users\dwerp\%%USERPROFILE%%)
 or the file path given by the 'CONDARC' environment variable, if it is set
 (default: %(default)s)."""
params = {"default": r"c:\Users\dwerp\%USERPROFILE%", ...}
y = x % params
y == r"""Write to the given file. Otherwise writes to the user config file (c:\Users\dwerp\%USERPROFILE%)
 or the file path given by the 'CONDARC' environment variable, if it is set
 (default: c:\Users\dwerp\%USERPROFILE%)."""
```

Reply:
Hence the contents of `default` will not be used as a string formatting template but will just be inserted.

Reply:
As for that user's messed up `USERPROFILE` setting, I'm pretty sure they have an invalid content in their (I think) `HOME` variable (thus my request for them to show the contents of that variables in https://github.com/conda/conda/issues/6644#issuecomment-355760312). That should potentially cause many more problems for them outside of `conda` anyways...

Reply:
> That should potentially cause many more problems for them outside of conda anyways...

Yeah you'd think so huh...

Reply:
Ok I think I got it right now.  I'm starting to get tired and wear down :-p

Reply:
Knowing that it must've been early in the morning in your timezone when you messaged me today, I'd say that's a pretty long day of work! Time to take a break ;)

Reply:
Yeah, time for a 🍺 or 🍻  :-D